state/migration_export.go

```diff
@@ -1889,30 +1892,83 @@ func (e *exporter) secrets() error {
 		consumersByID[id] = append(consumersByID[id], consumerArg)
 	}
 
+	allRemoteConsumers, err := store.allSecretRemoteConsumers()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	remoteConsumersByID := make(map[string][]description.SecretRemoteConsumerArgs)
+	for _, info := range allRemoteConsumers {
+		consumer, err := names.ParseTag(info.ConsumerTag)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		id := strings.Split(e.st.localID(info.DocID), "#")[0]
+		remoteConsumerArg := description.SecretRemoteConsumerArgs{
+			Consumer:        consumer,
+			CurrentRevision: info.CurrentRevision,
+		}
+		remoteConsumersByID[id] = append(remoteConsumersByID[id], remoteConsumerArg)
+	}
+
 	for _, md := range allSecrets {
 		owner, err := names.ParseTag(md.OwnerTag)
 		if err != nil {
 			return errors.Trace(err)
 		}
 		arg := description.SecretArgs{
-			ID:             md.URI.ID,
-			Version:        md.Version,
-			Description:    md.Description,
-			Label:          md.Label,
-			RotatePolicy:   md.RotatePolicy.String(),
-			Owner:          owner,
-			Created:        md.CreateTime,
-			Updated:        md.UpdateTime,
-			NextRotateTime: md.NextRotateTime,
-			Revisions:      revisionArgsByID[md.URI.ID],
-			ACL:            accessArgsByID[md.URI.ID],
-			Consumers:      consumersByID[md.URI.ID],
+			ID:              md.URI.ID,
+			Version:         md.Version,
+			Description:     md.Description,
+			Label:           md.Label,
+			RotatePolicy:    md.RotatePolicy.String(),
+			Owner:           owner,
+			Created:         md.CreateTime,
+			Updated:         md.UpdateTime,
+			NextRotateTime:  md.NextRotateTime,
+			Revisions:       revisionArgsByID[md.URI.ID],
+			ACL:             accessArgsByID[md.URI.ID],
+			Consumers:       consumersByID[md.URI.ID],
+			RemoteConsumers: remoteConsumersByID[md.URI.ID],
 		}
 		e.model.AddSecret(arg)
 	}
 	return nil
 }
 
+func (e *exporter) remoteSecrets() error {
```

Comment:
We should _really_ be using the established visitor pattern in order to add migrated entities. See _state/migrations_.